File: app/src/main/java/com/gazilla/mihail/gazillaj/utils/AppDialogs.java
package com.gazilla.mihail.gazillaj.utils;

import android.content.Context;
import android.content.DialogInterface;
import android.graphics.Bitmap;
import android.graphics.Color;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.gazilla.mihail.gazillaj.R;
import com.gazilla.mihail.gazillaj.presentation.main.card.CardView;
import com.gazilla.mihail.gazillaj.presentation.main.presentation.MainView;
import com.gazilla.mihail.gazillaj.ui.main.MainActivity;
import com.gazilla.mihail.gazillaj.utils.POJO.Notificaton;
import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.nostra13.universalimageloader.core.ImageLoader;
import com.nostra13.universalimageloader.core.ImageLoaderConfiguration;

public class AppDialogs {

    private AlertDialog alertDialog;

    public void errorDialog(Context context, String error, String location){

        LayoutInflater inflater = LayoutInflater.from(context);
        View viewDialog = inflater.inflate(R.layout.dialog_error, null);

        TextView textView= viewDialog.findViewById(R.id.tvErrorDialogError);
        textView.setText(error);

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);
        builder.setNegativeButton("Закрыть", (dialog1, which) -> {
            alertDialog.dismiss();
            alertDialog=null;
        });

        builder.setView(viewDialog);
        alertDialog = builder.create();
        alertDialog.show();

        Button bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE);
        bt.setTextColor(Color.rgb(254, 194, 15));

    }

    public void warningDialog(Context context, String warningMess){
        LayoutInflater inflater = LayoutInflater.from(context);
        View viewDialog = inflater.inflate(R.layout.dialog_error, null);

        TextView textView= viewDialog.findViewById(R.id.tvErrorDialogError);
        textView.setText(warningMess);

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);
        builder.setNegativeButton("Закрыть", (dialog1, which) -> {
            alertDialog.dismiss();
            alertDialog=null;
        });

        builder.setView(viewDialog);
        alertDialog = builder.create();
        alertDialog.show();

        Button bt2 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE);
        bt2.setTextColor(Color.rgb(254, 194, 15));
    }

    public void loadingDialog(Context context){
        LayoutInflater inflater = LayoutInflater.from(context);
        View viewDialog = inflater.inflate(R.layout.dialog_loading, null);

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);

        builder.setView(viewDialog);
        alertDialog = builder.create();
        alertDialog.show();
    }

    public void dialogWithQRCode(Context context, String dataForQrCode){
        String text = "Дождитесь, пока сотрудник\n" +
                "считает Ваш QR-код";
        try {
            LayoutInflater inflater = LayoutInflater.from(context);

            Bitmap bitmap = QRcode.encodeAsBitmap(dataForQrCode, BarcodeFormat.QR_CODE, 200, 200);

            View dialog = inflater.inflate(R.layout.dialod_with_qr_code, null);

            ImageView imgQRcode = dialog.findViewById(R.id.imvQRcodeDetailPresentDialog);
            imgQRcode.setImageBitmap(bitmap);

            TextView tv = dialog.findViewById(R.id.tvTxtDialogByFirst);
            tv.setText(text);

            AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);
            builder.setPositiveButton("Готово", (dialog1, which) -> {
                //presentPresenter.byPresent();
                //firstDialogWithQRcode.dismiss();
            })
                    .setNegativeButton("Отмена", (dialog12, which) -> alertDialog.dismiss());
            builder.setView(dialog);
            alertDialog = builder.create();
            alertDialog.show();

            Button bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE);
            Button bt1 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE);
            bt.setTextColor(Color.rgb(254, 194, 15));
            bt1.setTextColor(Color.rgb(254, 194, 15));

        } catch (WriterException e) {
            e.printStackTrace();
        }
    }

    public void dialogWinWheel(Context context, String win, String res, Boolean showBalanceTip, CardView cardView){

        ImageLoader imageLoader ;
        imageLoader = ImageLoader.getInstance();
        imageLoader.init(ImageLoaderConfiguration.createDefault(context));

        LayoutInflater inflater = LayoutInflater.from(context);
        View dialog = inflater.inflate(R.layout.dialog_win_wheel, null);

        ImageView winImg = dialog.findViewById(R.id.imgWinDialog);
        TextView tvWin= dialog.findViewById(R.id.tvWinDialog);

        imageLoader.displayImage(res, winImg);
        tvWin.setText(win);

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);

        builder.setPositiveButton("Спасибо!", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                if (showBalanceTip)
                    cardView.nextTip(2);
                alertDialog.dismiss();
                alertDialog=null;


            }
        });

        builder.setView(dialog);
        alertDialog = builder.create();
        alertDialog.show();

        Button bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE);
        bt.setTextColor(Color.rgb(254, 194, 15));
    }

    public void dialogFirstStart(Context context){
        LayoutInflater inflater = LayoutInflater.from(context);
        View dialog = inflater.inflate(R.layout.dialog_first_opennig, null);

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);

        builder.setPositiveButton("Продолжить!", (dialog1, which) -> {
            alertDialog.dismiss();
            alertDialog=null;
        });

        builder.setView(dialog);
        alertDialog = builder.create();
        alertDialog.show();

        Button bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE);
        bt.setTextColor(Color.rgb(254, 194, 15));
    }

    public void clouseDialog(){
        if (alertDialog!=null)
            alertDialog.dismiss();
    }

    public void dialogNotification(Context context, Notificaton notificaton, MainView mainView){
        LayoutInflater inflater = LayoutInflater.from(context);
        View dialog = inflater.inflate(R.layout.dialog_notification, null);

        TextView tvMessNotification = dialog.findViewById(R.id.tvMessNotification);
        tvMessNotification.setText(notificaton.getMessage());

        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.MyDialogTheme);
        builder.setNegativeButton("Закрыть", (dialog1, which) ->  {
            mainView.sendAnswerNotification(notificaton.getId(), -1);
            alertDialog.dismiss();
            alertDialog = null;

        });

        if (notificaton.getPromoId()!=0){
            builder.setPositiveButton("Акции", (dialog1, which) -> {
                mainView.openMenuStocks();
            });
        }
        else {
        if (notificaton.getCommands()!=null){
            if (notificaton.getCommands()[0] == 2 ){
                builder.setPositiveButton("Резерв", ((dialog1, which) -> {
                    alertDialog.dismiss();
                    alertDialog = null;
                    mainView.startReserveActivity();
                    mainView.sendAnswerNotification(notificaton.getId(), 2);
                }));
            }
            if (notificaton.getCommands()[0] == 3){
                builder.setPositiveButton("К колесу", (dialog1, which) -> {
                    alertDialog.dismiss();
                    alertDialog = null;
                    mainView.sendAnswerNotification(notificaton.getId(), 3);
                });
            }
            if (notificaton.getCommands()[0]==1||notificaton.getCommands()[0]==0){
                builder.setPositiveButton("Меню", (dialog1, which) -> {
                   mainView.openMenuPresent();
                    mainView.sendAnswerNotification(notificaton.getId(), 0);
                });
            }
        }
        }

        builder.setView(dialog);
        alertDialog = builder.create();
        alertDialog.show();

        Button bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE);
        bt.setTextColor(Color.rgb(254, 194, 15));

        Button bt1 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE);
        bt1.setTextColor(Color.rgb(254, 194, 15));
    }
}

File: app/src/main/java/com/gazilla/mihail/gazillaj/kotlin/helps/AppDialogs.kt
package com.gazilla.mihail.gazillaj.kotlin.helps

import android.content.Context
import android.content.DialogInterface
import android.graphics.Bitmap
import android.graphics.Color
import android.support.v7.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.gazilla.mihail.gazillaj.R
import com.gazilla.mihail.gazillaj.kotlin.pojo.Notification
import com.gazilla.mihail.gazillaj.kotlin.presenters.ProductPresenter
import com.gazilla.mihail.gazillaj.kotlin.views.CardView
import com.gazilla.mihail.gazillaj.kotlin.views.MainView
import com.gazilla.mihail.gazillaj.kotlin.views.ReserveView
import com.gazilla.mihail.gazillaj.kotlin.views.StartAppInitializationView
import com.squareup.picasso.Picasso

class AppDialogs {
    
    private lateinit var alertDialog: AlertDialog
    
    fun closeDialog(){
        if (alertDialog.isShowing)
            alertDialog.dismiss()
    }

    fun loadingDialog(context: Context) {
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_loading, null)

        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()
    }

    fun messageDialog(context: Context, error: String?) {

        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_error, null)

        val textView = viewDialog.findViewById<TextView>(R.id.tvErrorDialogError)
        textView.text = error

        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)
        builder.setNegativeButton("Закрыть") { _, _ ->
            alertDialog.dismiss()
        }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        val bt = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogWithQRCode(context: Context, bitmap: Bitmap, productPresenter: ProductPresenter){
        val infleter = LayoutInflater.from(context)
        val viewDialog = infleter.inflate(R.layout.dialod_with_qr_code, null)

        val imvQRCode = viewDialog.findViewById<ImageView>(R.id.imvQRcodeProductDialog)
        imvQRCode.setImageBitmap(bitmap)


        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)
        builder.setPositiveButton("Готово") { _ ,_ ->
            productPresenter.accessBuyProduct()
            alertDialog.dismiss()
        }.setNegativeButton("Отмена") { _, _ ->
            alertDialog.dismiss()
        }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.getButton(DialogInterface.BUTTON_POSITIVE).setTextColor(Color.rgb(254, 194, 15))
        alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE).setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogWinWheel(context: Context, res: Int, description: String){
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_win_wheel, null)

        val imgItemWin = viewDialog.findViewById<ImageView>(R.id.imgWinDialog)
        val descriptionInfo = viewDialog.findViewById<TextView>(R.id.tvWinDialog)

        Picasso.with(context).load(res).into(imgItemWin)

        descriptionInfo.text = description

        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)
        builder.setPositiveButton("Спасибо!"){ _, _ ->
            alertDialog.dismiss()
        }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.getButton(DialogInterface.BUTTON_POSITIVE).setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogDetailLvlDracon(context: Context, myLvlDiscription: String, one: String, two: String, fri: String, four: String, fif : String){
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_detail_progress, null)

        viewDialog.findViewById<TextView>(R.id.tvLVLDetailProgress).text = myLvlDiscription
        viewDialog.findViewById<TextView>(R.id.tvOneDetailProgress).text = one
        viewDialog.findViewById<TextView>(R.id.tvTwoDetailProgress).text = two
        viewDialog.findViewById<TextView>(R.id.tvThreeDetailProgress).text = fri
        viewDialog.findViewById<TextView>(R.id.tvFourDetailProgress).text = four
        viewDialog.findViewById<TextView>(R.id.tvFifDetailProgress).text = fif


        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)
        builder.setPositiveButton("Понятно!"){ _, _ ->
            alertDialog.dismiss()
        }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.getButton(DialogInterface.BUTTON_POSITIVE).setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogUpdateApp(context: Context, error: String?, startAppInitializationView : StartAppInitializationView) {

        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_error, null)

        val textView = viewDialog.findViewById<TextView>(R.id.tvErrorDialogError)
        textView.text = error

        val builder = AlertDialog.Builder(context, R.style.MyDialogTheme)
        .setPositiveButton("Обновить"){ _, _ ->
            startAppInitializationView.openPlayMarketForUpdate()
        }.setNegativeButton("Закрыть"){ _, _ ->
                    startAppInitializationView.dontUpdateApp()
                }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.setCancelable(false)

        val bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
        val bt2 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt2.setTextColor(Color.rgb(254, 194, 15))

    }

    fun dialogOpenReserve(context: Context, error: String?, cardView: CardView){
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_error, null)

        val textView = viewDialog.findViewById<TextView>(R.id.tvErrorDialogError)
        textView.text = error

        val builder = AlertDialog.Builder(context, R.style.StackedAlertDialogStyle)
                .setNegativeButton("Закрыть"){ _, _ ->
                    alertDialog.dismiss()
                }
                .setPositiveButton("Забронировать стол"){ _, _ ->
                    cardView.openReserveActivity()
                }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.setCancelable(false)

        val bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
        val bt2 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt2.setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogReserveSuccess(context: Context, message: String, reserveView: ReserveView){
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_error, null)

        val textView = viewDialog.findViewById<TextView>(R.id.tvErrorDialogError)
        textView.text = message

        val builder = AlertDialog.Builder(context, R.style.StackedAlertDialogStyle)
                .setNegativeButton("Спасибо"){ _, _ ->
                    alertDialog.dismiss()
                    reserveView.closeReserveActivity()
                }


        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.setCancelable(false)

        val bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
        val bt2 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt2.setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogDetailMyReserve(context: Context, message: String, reserveView: ReserveView ){
        val inflater = LayoutInflater.from(context)
        val viewDialog = inflater.inflate(R.layout.dialog_error, null)

        val textView = viewDialog.findViewById<TextView>(R.id.tvErrorDialogError)
        textView.text = message

        val builder = AlertDialog.Builder(context, R.style.StackedAlertDialogStyle)
                .setNegativeButton("Удалить"){ _, _ ->
                    reserveView.deleteMyReserve()
                    alertDialog.dismiss()
                }.setPositiveButton("Спасибо"){  _, _ ->
                    alertDialog.dismiss()
                }

        builder.setView(viewDialog)
        alertDialog = builder.create()
        alertDialog.show()

        val bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
        val bt2 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt2.setTextColor(Color.rgb(254, 194, 15))
    }

    fun dialogNotification(context: Context, notification: Notification, mainView: MainView, bitmap: Bitmap?) {
        val inflater = LayoutInflater.from(context)
        val dialog = inflater.inflate(R.layout.dialog_notification, null)

        val tvMessNotification = dialog.findViewById<TextView>(R.id.tvMessNotification)
        tvMessNotification.text = notification.message
        val imgNotification = dialog.findViewById<ImageView>(R.id.imgNotification)

        if (bitmap != null)
            imgNotification.setImageBitmap(bitmap)
        else
            imgNotification.visibility = View.GONE

        val builder = AlertDialog.Builder(context, R.style.StackedAlertDialogStyle)
        builder.setNegativeButton("Пропустить") { _, _ ->
            mainView.sendAnswerNotification(notification.id, -1)
            alertDialog.dismiss()
        }

        if (notification.promoId != 0) {
            builder.setPositiveButton("Акции") { _, _ ->
                mainView.openMenuFragmentBeforeNotification("StocksFragment")
            }
        }else{
            when{
                notification.commands[0] == 1 || notification.commands[0] == 0-> {
                    builder.setPositiveButton("Меню") { _, _ ->
                        mainView.openMenuFragmentBeforeNotification("ProductsMenuFragment")
                        mainView.sendAnswerNotification(notification.id, 0)
                    }
                }
                notification.commands[0] == 2 -> {
                    builder.setPositiveButton("Забронировать") { _, _ ->
                        mainView.startReserveActivity()
                        mainView.sendAnswerNotification(notification.id, 2)
                    }
                }
                notification.commands[0] == 3 -> {
                    builder.setPositiveButton("К колесу") { _, _ ->
                        mainView.sendAnswerNotification(notification.id, 3)
                    }
                }
            }
        }

        builder.setView(dialog)
        alertDialog = builder.create()
        alertDialog.show()

        alertDialog.setCanceledOnTouchOutside(false)

        val bt = alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)
        bt.setTextColor(Color.rgb(254, 194, 15))
        val bt1 = alertDialog.getButton(DialogInterface.BUTTON_NEGATIVE)
        bt1.setTextColor(Color.rgb(254, 194, 15))
    }
    
}